ls1intum/ArTEMiS
# Allow tutors to adapt their assessments

### Checklist

### Motivation and Context


Tutors want to adapt their (modeling and text) assessments after they submitted it.

### Description

We allow tutors to override their own assessments before the assessment due date. Instructors can still override any assessment at any time. Additionally, we only show assessments to students if the assessment due date of the corresponding exercise is already over. If there is no assessment due date, tutors are not allowed to override their assessments and students get the assessments right after it is done.

### Steps for Testing


1. Log in to ArTEMiS as a tutor
2. Navigate to tutor dashboard
3. Assess modeling/text submission of exercise with assessment due date in the future
4. Override assessment
5. Try the same for an exercise with assessment due date in the past or no assessment due date -> override does not work
5. Log in to ArTEMiS as a student
6. Go to course overview
7. See results for exercise with assessment due date in the past
8. Result for exercise with assessment due date in the future are not displayed
